Cargo.toml

```diff
@@ -28,12 +28,13 @@ tracing = { version = "0.1.36" }
 # this sections is shared by tests, benchmarks, and examples
 [dev-dependencies]
 criterion = { version = "0.3", features = ["html_reports"] }
-rand = "0.7"
-rand_core = "0.5"
+rand = "0.8"
+rand_core = "0.6"
 
 [[bench]]
 harness = false
 name = "blitzar_benchmarks"
+required-features = ["rand_core"]
```

Comment:
I don't think this is needed if `rand_core` gets added to the `dev-dependencies`

Reply:
I added `curve25519-dalek = { version = "4", features = ["rand_core"] }` to `dev-dependencies`, but removing this line produced the same error as before I added the `required-feature` line.

![image](https://github.com/spaceandtimelabs/blitzar-rs/assets/6509385/a2b58edd-a61f-4b71-ad8b-19397f8bcf71)

Reply:
Try removing the `&` so that it just looks like `.map(|_| (&Scalar::random(&mut rng) * constants::RISTRETTO_BASEPOINT_TABLE))`.
I'm not sure why the `rand_core` feature has anything to do with that, though.

Reply:
That was the missing piece! Updated and removed. Thanks.